modules/sync/src/main/kotlin/dev/proxyfox/sync/PkSync.kt

```diff
@@ -8,26 +8,76 @@
 
 package dev.proxyfox.sync
 
+import com.google.common.collect.HashBiMap
+import dev.proxyfox.common.annotations.DontExpose
+import dev.proxyfox.common.throwIfPresent
+import dev.proxyfox.database.database
 import dev.proxyfox.database.records.system.SystemRecord
 import dev.proxyfox.pluralkt.PluralKt
-import dev.proxyfox.pluralkt.types.PkSystem
+import dev.proxyfox.pluralkt.Response
+import dev.proxyfox.pluralkt.types.PkError
+import dev.proxyfox.pluralkt.types.PkMember
 
 object PkSync {
-    fun pull(token: String, system: SystemRecord) {
-        PluralKt.System.getMe(token) {
-            if (isSuccess()) {
-                getSuccess().getMembers(token)
-            }
-        }
-    }
+    private fun <T> Response<T>.getSuccessOrNull() = if (isSuccess()) getSuccess() else null
+
+    @OptIn(DontExpose::class)
+    suspend fun pull(system: SystemRecord): PkError? {
+        val token = system.pkToken ?: return null
+
+        val systemResp = PluralKt.System.getMe(token).await()
```

Comment:
Consideration, we might want to pass ProxyFox's UA through, maybe compounding on PluralKt's UA.